hwmv2: convert NXP S32 SoC and boards to hardware model v2

manuargue opened the pull request:
Convert NXP S32 SoC and boards to hardware model v2. Depends on [hal_nxp#340.](https://github.com/zephyrproject-rtos/hal_nxp/pull/340)

manuargue-nxp committed:
boards: nxp: ucans32k1sic: convert to hwmv2

Convert ucans32k1sic board to hardware model v2.

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

manuargue-nxp committed:
boards: nxp: mr_canhubk3: convert to hwmv2

Convert mr_canhubk3 board to hardware model v2.

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

manuargue-nxp committed:
soc: nxp: convert NXP S32 family to hwmv2

Convert NXP S32 family to hardware model v2.

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

nordicjm commented on soc/nxp/s32/s32k1/Kconfig.soc:
```diff
@@ -1,65 +1,56 @@
+	select SOC_FAMILY_NXP_S32
+
+config SOC_SERIES
+	default "s32k1" if SOC_SERIES_S32K1XX
```
```suggestion
	default "s32k1xx" if SOC_SERIES_S32K1XX
```

nordicjm replied:
Thanks, have created an issue: https://github.com/zephyrproject-rtos/zephyr/issues/69317

nordicjm commented on soc/nxp/s32/s32k3/Kconfig.soc:
```diff
@@ -0,0 +1,24 @@
+	select SOC_FAMILY_NXP_S32
+
+config SOC_SERIES
+	default "s32k3" if SOC_SERIES_S32K3XX
```
```suggestion
	default "s32k3xx" if SOC_SERIES_S32K3XX
```

nordicjm commented on soc/nxp/s32/soc.yml:
```diff
@@ -0,0 +1,22 @@
+family:
+- name: nxp_s32
+  series:
+  - name: s32k1
```
```suggestion
  - name: s32k1xx
```

nordicjm commented on soc/nxp/s32/soc.yml:
```diff
@@ -0,0 +1,22 @@
+    - name: s32k144w
+    - name: s32k146
+    - name: s32k148
+  - name: s32k3
```
```suggestion
  - name: s32k3xx
```

manuargue-nxp committed:
boards: nxp: s32z2xxdc2: convert to hwmv2

Convert `s32z270dc2` boards to hardware model v2. The board has been
renamed to `s32z2xxdc2` to be able to support in the future other
SoCs from this series that can also work on this board.

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>

iuliana-prodan commented on soc/nxp/s32/Kconfig.defconfig:
```diff
@@ -0,0 +1,8 @@
+
+rsource "*/Kconfig.defconfig"
+
+endif # SOC_FAMILY_KINETIS
```
s/SOC_FAMILY_KINETIS/SOC_FAMILY_NXP_S32

manuargue-nxp committed:
tests: drivers: gpio: gpio_api_1pin: exclude hifive1

Test seems to be failing consistently on `hifive1` board.
See #69350

Signed-off-by: Manuel Argüelles <manuel.arguelles@nxp.com>
